zingolib/src/wallet/transaction_context.rs

```diff
@@ -397,80 +468,27 @@ pub mod decrypt_transaction {
             .await;
         }
 
-        /// Transactions contain per-protocol "bundles" of components.
-        /// The component details vary by protocol.
-        /// In Sapling the components are "Spends" and "Outputs"
-        /// In Orchard the components are "Actions", each of which
-        /// _IS_ 1 Spend and 1 Output.
-        #[allow(clippy::too_many_arguments)]
-        async fn decrypt_transaction_to_record_domain<D: DomainWalletExt>(
+        async fn account_for_shielded_receipts<Q>(
             &self,
-            transaction: &Transaction,
+            ivk: Ivk<Q, External>,
+            domain_tagged_outputs: &[(
+                Q,
+                <<Q as DomainWalletExt>::Bundle as wallet::traits::Bundle<Q>>::Output,
```

Comment:
everywhere else is the codebase Domain is represented by D so i find this more confusing to read with Q, im not sure why we would change it